brave/src/main/java/brave/internal/handler/MutableSpanConverter.java

```diff
@@ -65,38 +76,30 @@ void convert(MutableSpan span, Span.Builder result) {
         .build());
     }
 
-    String errorTag = span.tag("error");
-    if (errorTag == null && span.error() != null) {
-      errorParser.error(span.error(), span);
-    }
-
     span.forEachTag(Consumer.INSTANCE, result);
     span.forEachAnnotation(Consumer.INSTANCE, result);
     if (span.shared()) result.shared(true);
+    if (span.debug()) result.debug(true);
   }
 
   // avoid re-allocating an endpoint when we have the same data
-  void addLocalEndpoint(String serviceName, @Nullable String ip, int port, Span.Builder span) {
-    if (serviceName == null) serviceName = localServiceName;
-    if (ip == null) ip = localIp;
-    if (port <= 0) port = localPort;
-    if (localServiceName.equals(serviceName)
-      && (localIp == null ? ip == null : localIp.equals(ip))
-      && localPort == port) {
-      span.localEndpoint(localEndpoint);
+  void addLocalEndpoint(@Nullable String serviceName, @Nullable String ip, int port, Builder span) {
+    if (serviceName != null) serviceName = serviceName.toLowerCase(Locale.ROOT);
+    if (hashEndpointParameters(serviceName, ip, port) == defaultEndpointHashCode) {
+      span.localEndpoint(defaultEndpoint);
     } else {
       span.localEndpoint(Endpoint.newBuilder().serviceName(serviceName).ip(ip).port(port).build());
     }
   }
 
-  enum Consumer implements TagConsumer<Span.Builder>, AnnotationConsumer<Span.Builder> {
+  enum Consumer implements TagConsumer<Builder>, AnnotationConsumer<Builder> {
```

Comment:
Possibly was due to refactoring - suggest putting Span back here

Reply:
it was line wrap, but yeah ok.